pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -171,31 +201,51 @@ func createGatewayHTTPFilterChainOpts(
 	}
 
 	httpListeners := make([]*filterChainOpts, 0, len(servers))
-	for i, server := range servers {
-		routeCfg := buildGatewayInboundHTTPRouteConfig(env, nameToServiceMap, gatewayNames, server)
-		if routeCfg == nil {
-			log.Debugf("omitting HTTP listeners for port %d filter chain %d due to no routes", server.Port, i)
-			continue
-		}
-		// if https redirect is set, we need to enable requireTls field in all the virtual hosts
-		if server.Tls != nil && server.Tls.HttpsRedirect {
-			for i := range routeCfg.VirtualHosts {
-				// TODO: should this be set to ALL ?
-				routeCfg.VirtualHosts[i].RequireTls = route.VirtualHost_EXTERNAL_ONLY
-			}
-		}
+	// Are we processing plaintext servers or TLS servers?
+	// If plain text, we have to combine all servers into a single listener
+	if model.ParseProtocol(servers[0].Port.Protocol) == model.ProtocolHTTP {
+		rdsName := model.GatewayRDSRouteName(servers[0])
+		routeCfg := buildGatewayInboundHTTPRouteConfig(env, nameToServiceMap, gatewayNames, servers, rdsName)
 		o := &filterChainOpts{
-			sniHosts:   getSNIHosts(server),
-			tlsContext: buildGatewayListenerTLSContext(server),
+			// This works because we validate that only HTTPS servers can have same port but still different port names
+			// and that no two non-HTTPS servers can be on same port or share port names.
+			// Validation is done per gateway and also during merging
+			sniHosts:   nil,
+			tlsContext: nil,
 			httpOpts: &httpListenerOpts{
 				routeConfig:      routeCfg,
```

Comment:
Is this going to be removed with RDS used? I understand the sidecar has to be configured to use RDS. Guess it's going to be done later?